docs/release.md

```diff
@@ -0,0 +1,180 @@
+The change to any production cluster implied by a new release varies. It may be nothing; alternatively it is also feasible that many or all versioned items could change in a release.
+We aim to ensure that during normal operations, clusters are created at, and upgraded to, versions exactly corresponding with a released manifest according to the cluster’s stream.
+Irrespective of the stream, we aim for the spread of production cluster versions to be as small as feasible (e.g. n.0, (n-1).0, (n-1).1, (n-2).0).  Exact spread goal TBD.  A larger spread implies more testing.
+
+Within a stream, we aim to be able to directly upgrade any deployed production cluster to the latest version.
+Versioning
+
+## Major releases
+The plugin major release (“n”) is incremented with each weekly release.
+
+### Major release creation flow:
+
+1. Create a branch `release-n` with a plugin code you wish to release. It is preferable, but not essential, that we branch master. We expect the commit that is being branched to be passing tests
+2. Configuration step for CI so new release branch would be tested with our test suites
+
+Follow minor release flow
+The above should be automated as much as possible, e.g. pushing a git tag triggers image builds
+
+## Minor releases
+
+### Minor release creation flow (“n.x”):
+
+1. Make a commit in `release-n`:
+   in each stream’s manifest we set the tag of each openshift-azure container image to `:vn.x`
+   in each stream’s manifest we set the plugin release version to `vn.x`
+
+2. Git tag release (`vn.x`)
+3. Build openshift-azure container image (`sync:vn.x`, etc)
+4. If the minor release is a cherry-pick (not a straightforward branch of master), carry out update testing before release to MSFT
+5. (TBD) CI test on the result? User documentation release procedure? Release procedure to MSFT?
+
+Branching model
+```
+-------master--*----------*---------------------->
+               \          \
+                \           \-release-v4----T(v4.0)-----T(v4.1)->
+                 \-release-v3------T(v3.0)---->
+```
+
+
+# Doing a release with current integration:
+
+1. Create a branch from master and push it. This will create branch for us to do release in.
+
+```
+git checkout <required commit> 
+git checkout -b release-vx
+# we create release branch for first weekly only
+git push upstream release-vx
+```
+
+If it is minor release:
+```
+git checkout release-vx
+git checkout -b release-vx.y-fix
+git cherry-pick <git commit id>
+```
+
+Open a PR into release branch.
+
+2. Configure testing for release branch in `openshift/release` repository
+
+```
+git checkout master
+git fetch upstream 
+git rebase upstream/master 
+git checkout -b osa.vx.y.release
+```
+
+Add new branch config file in `ci-operator/config/openshift/openshift-azure/openshift-openshift-azure-release-vx.yaml`
+
+Add new test jobs for release branch:
+`ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-release-vx-presubmits.yaml`
```

Comment:
at the moment our release branch presubmits seem to disable always_run on the images target, and enable it on lots of the e2e tests.  Needs to be documented.  wrt the images target, is this intentional and why?

Reply:
It was intentional, but at the same time, it does not make any difference because of how graph is being built in ci-operator.